### PR TITLE
[BugFix] fix show materialized view on follower node

### LINKED ISSUE #30052: Execute show materialized view got error on follower FE
### Steps to reproduce the behavior (Required)

Execute such command on follower FE:
```sql
mysql > show materialized views;
(1064, 'Socket is closed by peer.')
```

### StarRocks version (Required)
- 2.5, 3.0, 3.1

### PR DESCRIPTION
**Pull request by murphyatwork**
Fixes #30052

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowResultSet.java [hunk 1]
```diff
@@ -129,9 +130,8 @@ public TShowResultSet tothrift() {
         }
 
         set.resultRows = Lists.newArrayList();
-        for (int i = 0; i < resultRows.size(); i++) {
-            ArrayList<String> list = Lists.newArrayList();
-            list.addAll(resultRows.get(i));
+        for (List<String> resultRow : resultRows) {
+            List<String> list = resultRow.stream().map(x -> Objects.toString(x, "")).collect(Collectors.toList());
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowResultSet.java:134]: why resultRows.get(i) will be null?

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowResultSet.java:134]: There's no explicit limitation for null.

For `show materialized views` command, the `inactive_reason` field may be null.

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 0 / 2 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ShowResultSet.java|0|2|00.00%|[133, 134]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 0 / 2 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ShowResultSet.java|0|2|00.00%|[133, 134]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 0 / 2 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ShowResultSet.java|0|2|00.00%|[133, 134]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 0 / 2 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ShowResultSet.java|0|2|00.00%|[133, 134]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**